test/junit/scala/collection/immutable/NumericRangeTest.scala

```diff
@@ -32,4 +32,26 @@ class NumericRangeTest {
     val x = (BigDecimal(1) to BigDecimal(3) by 1).sum
     assertEquals(BigDecimal(6), x)
   }
+
+  @Test
+  def t11152_BigDecimalMakesProgress: Unit = {
+    // Overflow case with default MathContext
+    val a = BigDecimal(1)
+    val b = a + BigDecimal("1e-30")
+    val c = BigDecimal("1e-38")
+    assertTrue{
+      try{ (a to b by c).length >= 0 }  // Force evaluation of `length`
+      catch {
+        case _: IllegalArgumentException => true  // This should be thrown
```

Comment:
If I'm understanding this correctly, the assert succeeds in all cases, because `length` cannot be negative. If so, what is it testing?

Reply:
@NthPortal - It is testing that it throws an `IllegalArgumentException` if `c` is so small that it is lost when added to `a` and/or `b`.  (Just like `(0 to 10 by 0).length` throws an exception.)

Reply:
the problem is, `(a to b by c).length` can never be negative, so the try block always returns `true` if it doesn't throw anyway

Reply:
Oops, how silly!  I meant `< 0` and somehow even though you said exactly what was wrong it took me two tries to notice that I didn't write what I meant.

Reply:
Fixed it.